src/llama_stack/core/server/tracing.py

```diff
@@ -33,6 +54,44 @@ async def __call__(self, scope, receive, send):
             logger.debug(f"Bypassing custom routing for FastAPI built-in path: {path}")
             return await self.app(scope, receive, send)
 
+        # Check if this is a router-based route - if so, pass through to FastAPI
+        # Router-based routes are handled by FastAPI directly, so we skip the old route lookup
+        # but still need to set up tracing
+        is_router_based = self._is_router_based_route(path)
```

Comment:
this feels like it should be a separate PR  just because there's already a lot of important stuff to focus on in this PR and so there cannot be enough focus / testing on this piece (which can be important because telemetry can be wrong if we don't get this right)